pantheon-bundle/frontend/src/app/contexts/ProductContext.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { createContext, useState, useEffect } from "react";
+
+export let ProductContext = createContext<IProduct[]>([] as IProduct[]);
```

Comment:
`ProductContext` was created as a way to share `allProducts` between `ProductDetails` and `ProductListing`